spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -0,0 +1,161 @@
+      end
+
+      it "Exemptions are populated" do
+        expect(xml.css('Exemptions').count).to eq(1)
+      end
+
+      it "populates line 6 XML fields" do
+        expect(xml.at("Exemptions YouRegular").text).to eq("X")
+        expect(xml.at("Exemptions SpouseCuRegular")).to eq(nil)
+        expect(xml.at("Exemptions DomesticPartnerRegular")).to eq(nil)
+      end
+
+      context "when filer is over 65" do
+        let(:intake) { create(:state_file_nj_intake, :primary_over_65) }
+        it "populates line 7 XML fields" do
+          expect(xml.at("Exemptions YouOver65").text).to eq("X")
+          expect(xml.at("Exemptions SpouseCuPartner65OrOver")).to eq(nil)
+        end
+      end
+
+      context "when filer is younger than 65" do
+        it "populates line 7 XML fields" do
+          expect(xml.at("Exemptions YouOver65")).to eq(nil)
+          expect(xml.at("Exemptions SpouseCuPartner65OrOver")).to eq(nil)
+        end
+      end
+    end
+
+    context "when filer is married filing jointly" do
+      let(:intake) { create(:state_file_nj_intake, :married_filing_jointly) }
+      it "Exemptions are populated" do
+        expect(xml.css('Exemptions').count).to eq(1)
+      end
+
+      it "populates line 6 XML fields" do
+        expect(xml.at("Exemptions YouRegular").text).to eq("X")
+        expect(xml.at("Exemptions SpouseCuRegular").text).to eq("X")
+        expect(xml.at("Exemptions DomesticPartnerRegular")).to eq(nil)
+      end
+
+      context "when filer is over 65 and spouse is under 65" do
+        let(:intake) { create(:state_file_nj_intake, :primary_over_65, :married_filing_jointly) }
+        it "populates line 7 XML fields" do
+          expect(xml.at("Exemptions YouOver65").text).to eq("X")
+          expect(xml.at("Exemptions SpouseCuPartner65OrOver")).to eq(nil)
+        end
+      end
+
+      context "when filer is over 65 and spouse is over 65" do
+        let(:intake) { create(:state_file_nj_intake, :primary_over_65, :mfj_spouse_over_65) }
+        it "populates line 7 XML fields" do
+          expect(xml.at("Exemptions YouOver65").text).to eq("X")
+          expect(xml.at("Exemptions SpouseCuPartner65OrOver").text).to eq("X")
+        end
+      end
+
+      context "when filer is under 65 and spouse is under 65" do
+        let(:intake) { create(:state_file_nj_intake, :married_filing_jointly) }
+        it "populates line 7 XML fields" do
+          expect(xml.at("Exemptions YouOver65")).to eq(nil)
+          expect(xml.at("Exemptions SpouseCuPartner65OrOver")).to eq(nil)
+        end
+      end
+
+      context "when filer is under 65 and spouse is over 65" do
+        let(:intake) { create(:state_file_nj_intake, :mfj_spouse_over_65) }
+        it "populates line 7 XML fields" do
+          expect(xml.at("Exemptions YouOver65")).to eq(nil)
+          expect(xml.at("Exemptions SpouseCuPartner65OrOver").text).to eq("X")
+        end
+      end
```

Comment:
[DUST] imo it might be overkill to test all 4 of these scenarios? i think it just needs one test each for the checked and unchecked case for each box. that said, i'm never really too fussed about more tests